Anthony/fix badges earned icon and blue square icon on profile page

AnthonyWeathers opened the pull request:
# Description
Made it so both Badge and Blue Square icons on the summary bar with your logged hours, having correct number displayed on their red circle notification of more new instances since you last clicked on the respective icon.

## Related PRS (if any):
This frontend PR is related to the backend [PR#1504](https://github.com/OneCommunityGlobal/HGNRest/pull/1504)
…

## Main changes explained:
- Updated Badge.jsx, mostly to add background color to highlight your total badges, and personal best of hours in a week, assuming you have that badge.
- Updated Badge.test.js, to work for the modified text I had modified in Badge.jsx
- Updated SummaryBar.jsx to hide the red circle when there are no new badges, or blue squares, or even task count for the icons so there's not a constant red circle for 0 count. Modified the Blue Square to be a button so its response to hover is the same as the other icons, and used history to perform page redirect. Also modified what value is used for Badge and Blue Square to track the number of added badges and blue squares.
- Updated SummaryBar.css to add styling for the white border behind the icons on hover
- Updated BlueSquare.jsx by adding an id and logic for the element referenced in the history push, so the redirect takes you to the blue square section of the user profile
…

## How to test:
1. check into current branch
2. do `npm install` and `...` to run this PR locally
3. Clear site data/cache
4. log as admin/owner user
5. go to dashboard
6. verify if no red circles are shown for some of the icons that do have them (note: for Blue Square, new value added/used is default 0, so you may not have a red notification for blue square on logging in for this PR)
7. Test Icons that they have a white circle border appear behind them when hovered over
8. If Badge icon has a red notification, verify if clicking on it, and opening the badge tab, clears the notification, after some time at the latest
9. Verify if the text with the colored backgrounds are visible
10. Then go to Profile and add a Blue Square to yourself (and a badge if you didn't have one to test earlier), then go back to Dashboard, and verify if the red notifications with 1 or however many you added, appear correctly, and verify if clicking the icons do cause the notifications to reset.
11. Also verify if clicking the Blue Square Icon scrolls you down to the Blue Square section, some space above/below is fine.
12. If the Blue Square icon does not get a red notification after adding a blue square to your own account, then check your .env file on the frontend, and ensure REACT_APP_APIENDPOINT is set to http://localhost:4500/api instead of https://api-staging.highestgood.com/api. If not set to http://localhost:4500/api, change the value to this and try again as https://api-staging.highestgood.com/api seems to not be working.

## Screenshots or videos of changes:

https://github.com/user-attachments/assets/b97f9455-8fb6-42f7-97fb-2c66219da974


## Note:
For the Blue Square number tracker, if you add blue squares, and then delete older ones, they will decrement the count, so if you add 2 then delete 2 older ones, going to dashboard will appear as if no new ones were added. Weren't sure whether to have a decrement at all, but can remove if its better to just have increment and reset. 

Other note, I do not know why, but if you go to user profile with the hash of blue square, so it scrolls you to the blue square section, if you enter the url again, it seems to remove the header part of the page, but a reload or back button, or delete the #bluesquare part of the url before pressing enter, restores the header.

suparshwa31 commented:
The changes are working,i can see white border when I hover over the badge and other buttons. as I didn't had any badges for my owner account it did not show any badges but everything else worked fine.
https://github.com/user-attachments/assets/61731385-b24f-4923-aec2-477b2eb86f5b

kanishkagarwal6101 reviewed (dismissed):
I can see the white border on the icons when hovered and when clicked on blue square icon it redirects and scrolls down to the appropriate view. 
<img width="1907" height="448" alt="Screenshot 2025-08-07 at 1 00 07 PM" src="https://github.com/user-attachments/assets/e7a050dc-41b5-4fa7-83fb-a2a732b10b0d" />

Yi9569 commented:
### Requesting Changes
1. I tested using admin account and added 3 blue squares to its own account, but on dashboard the blue square logo does not show numbered alerts
<img width="1470" height="871" alt="blue_square_no_alert" src="https://github.com/user-attachments/assets/db88f41f-398d-482a-a76b-380b0eadf56d" />

2. I have added 2 blue squares to the volunteer account, while they did get added as shown at the blue square page, there is no alert on the dashboard page
<img width="1470" height="851" alt="volunteer_no_alert" src="https://github.com/user-attachments/assets/104a278e-04b9-4269-a427-88b584735a9b" />
<img width="1470" height="855" alt="volunteer_blue_square_page_update_successful" src="https://github.com/user-attachments/assets/7cb28d39-3cb3-4c77-a1d9-2cc0017f3616" />

AnthonyWeathers commented:
> ### Requesting Changes
> 1. I tested using admin account and added 3 blue squares to its own account, but on dashboard the blue square logo does not show numbered alerts
> 
> <img alt="blue_square_no_alert" width="1470" height="871" src="https://private-user-images.githubusercontent.com/97994143/475848801-db88f41f-398d-482a-a76b-380b0eadf56d.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTQ3MTU2ODAsIm5iZiI6MTc1NDcxNTM4MCwicGF0aCI6Ii85Nzk5NDE0My80NzU4NDg4MDEtZGI4OGY0MWYtMzk4ZC00ODJhLWE3NmItMzgwYjBlYWRmNTZkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA4MDklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwODA5VDA0NTYyMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTFiZmQ3MWUxYzA4NGQ0MjU5YTYwNDNlMjEyMGMyYmIxOWQ5NDNjMDUzM2E5MWEzNTA3NDYyNmIwOGVmNjkwMzcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.4O4qKVWGWexvGJgcUiD3MfCdeL_-jLk75SvaBEUJqL8">
> 2. I have added 2 blue squares to the volunteer account, while they did get added as shown at the blue square page, there is no alert on the dashboard page
> 
> <img alt="volunteer_no_alert" width="1470" height="851" src="https://private-user-images.githubusercontent.com/97994143/475849862-104a278e-04b9-4269-a427-88b584735a9b.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTQ3MTU2ODAsIm5iZiI6MTc1NDcxNTM4MCwicGF0aCI6Ii85Nzk5NDE0My80NzU4NDk4NjItMTA0YTI3OGUtMDRiOS00MjY5LWE0MjctODhiNTg0NzM1YTliLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA4MDklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwODA5VDA0NTYyMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTExODc4MTg0MGFkYmIxMTg0MWRiNDE0YzIwNTUzNTViZmRhYzgzNzMwMWZlODYyMWE4YTBiOTI2YTk5ZDVmMjImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.sUd0L2h3iNNUY1r2QYzO9vVfa9MNB-KxqPj-cAwfZbQ"> <img alt="volunteer_blue_square_page_update_successful" width="1470" height="855" src="https://private-user-images.githubusercontent.com/97994143/475849863-7cb28d39-3cb3-4c77-a1d9-2cc0017f3616.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTQ3MTU2ODAsIm5iZiI6MTc1NDcxNTM4MCwicGF0aCI6Ii85Nzk5NDE0My80NzU4NDk4NjMtN2NiMjhkMzktM2NiMy00Yzc3LWExZDktMmNjMDAxN2YzNjE2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA4MDklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwODA5VDA0NTYyMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTIyZjI0NDQ5M2VlNGZkYTkzMjE5Mjg1Y2ViZTk0N2JiYzlhMzE4MjBmOWEyMzEyY2Q4MmM1YTE4YzRhNzY3NzMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.MO6s690h-UQRWRVG7e4RSAPWis3IZa3B_w8MRrZSXrY">

Hello there, I've been looking into the issue and encountered myself during testing, double checking my code and the logic flow, even the url to ensure I hadn't messed up with a recent git merge. Then I decided to modify the REACT_APP_APIENDPOINT value in env, I originally had it as REACT_APP_APIENDPOINT="https://api-staging.highestgood.com/api, which was how we were instructed to do so in [Steps to run the local setup](https://docs.google.com/document/d/1d-ZzDnb_OOXXUN49V-wExHpAgOFMYpe2se1SaziA7Xw/edit?usp=sharing), but after I changed it to "http://localhost:4500/api", it ended up actually triggering the backend, as I had setup console logs at the function to add infringements, which hadn't been getting logged until changing the endpoint to that.

If you could check and test it on your end to verify if my discovery is true, then next steps will be easier to be discussed or brought up in the slack channels.

donaldmucharla commented:
Hi @AnthonyWeathers 
<img width="1902" height="1008" alt="Screenshot 2025-08-15 111119" src="https://github.com/user-attachments/assets/23bfa8bd-d85f-4c06-8ab0-cd798419c915" />
<img width="1890" height="925" alt="Screenshot 2025-08-15 111553" src="https://github.com/user-attachments/assets/cf4a6142-41ed-4f1c-91d6-a2bf5dc1d182" />
, I tested using an admin account by adding 3 blue squares to my own account and also 2 to a volunteer account — in both cases, while the blue squares were successfully added and visible on the blue square page, no numbered alerts appeared on the dashboard icons. After double-checking my code, logic flow, and recent merges, I tested by changing REACT_APP_APIENDPOINT in the .env file from https://api-staging.highestgood.com/api (as per the local setup instructions) to http://localhost:4500/api. This change immediately triggered the backend as expected — confirmed by my console logs at the add infringements function, which previously were not firing. Could you please test on your end to confirm if this behavior is consistent? If so, it will help us narrow down the next steps.

donaldmucharla commented:
<img width="1891" height="956" alt="image" src="https://github.com/user-attachments/assets/0516eec4-5256-451d-8e59-13aa4aae688c" />
<img width="1900" height="955" alt="image" src="https://github.com/user-attachments/assets/51b4a7b6-94fe-4821-80d8-a1a5ef5fca29" />


Hi @AnthonyWeathers , I tried replicating the issue again and walked through my code logic, console logs, and even verified the routes after pulling the latest merge. Initially, the backend calls weren’t firing, but once I switched REACT_APP_APIENDPOINT in my .env from

https://api-staging.highestgood.com/api
to
http://localhost:4500/api


the logs started showing up as expected. That seems to be the key difference when testing locally. Could you confirm if the same works on your setup? That way we’ll know if the behavior is consistent.

Also, while testing with the Admin account, I added 3 blue squares, and although they showed in the profile, the dashboard notification count didn’t update. The hover effect still works, and clicking the blue square icon correctly scrolls to the section, but the alert number stays missing.

AnthonyWeathers commented:
> <img alt="Screenshot 2025-08-30 at 10 39 36 PM" width="1678" height="580" src="https://private-user-images.githubusercontent.com/35608051/483926933-2877800a-6b59-43bb-9452-dde9fe892216.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTY2OTM0MTEsIm5iZiI6MTc1NjY5MzExMSwicGF0aCI6Ii8zNTYwODA1MS80ODM5MjY5MzMtMjg3NzgwMGEtNmI1OS00M2JiLTk0NTItZGRlOWZlODkyMjE2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA5MDElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwOTAxVDAyMTgzMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTliNWVjZWY0ZTNjMTU1NGYwODRmMTVmOTU3ZTE0OGFkMTllZWI3YjhlYTEwMzJmYTUwZDVlNzQyYWViMGI0ZGQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.w-qRVVU12z8bhchRT_Q2n2oAyyaeLQq7Niz85X-4tCA">
> I added two Blue Squares to myself, but I was not able to see the notification of the Blue square on the dashboard.
> 
> But everything else that is mentioned in the PR description worked properly.

Hi, @vivekchandrabs, just to check but did you try this PR with REACT_APP_APIENDPOINT in the .env on the frontend set to
http://localhost:4500/api instead of https://api-staging.highestgood.com/api...? At some point, the endpoint we were given on the local setup document stopped working, so to actually get the backend working, I think its needed to have the value set as this in your .env file: REACT_APP_APIENDPOINT="http://localhost:4500/api".

If you do have this value setup, and it still doesn't update the infringmentCount, let me know.

Anusha-Gali reviewed (approved):
Hi Anthony,

I have reviewed your PR locally. Although the whole functionality works as expected. I just have one concern, why do we have to click again on the Badges in the Tasks and Timelogs section for the notifications to disappear, why does the count not disappear when we click on the Badges Earned logo which then open the Badges in the Tasks and Timelogs section?
**Badges**
<img width="484" height="398" alt="Screenshot 2025-11-18 at 7 22 05 PM" src="https://github.com/user-attachments/assets/d4306d2a-3175-4c1e-99f8-42afea0e70fe" />

<img width="484" height="398" alt="Screenshot 2025-11-18 at 7 22 13 PM" src="https://github.com/user-attachments/assets/6853520a-b33d-402e-9246-73dc5fd15b84" />

<img width="1267" height="752" alt="Screenshot 2025-11-18 at 7 22 29 PM" src="https://github.com/user-attachments/assets/bcda6dba-3143-49bc-9cc4-ace56eaad65a" />

<img width="733" height="483" alt="Screenshot 2025-11-18 at 7 27 55 PM" src="https://github.com/user-attachments/assets/045849c2-c656-48e5-a8a9-7ec305fa1e4f" />

**Blue Square**
<img width="421" height="118" alt="Screenshot 2025-11-18 at 7 19 08 PM" src="https://github.com/user-attachments/assets/bb4a232a-9bdc-4c8e-9988-125d472ba3ff" />

<img width="1240" height="609" alt="Screenshot 2025-11-18 at 7 20 22 PM" src="https://github.com/user-attachments/assets/dafd4b27-fe76-4d6e-8f55-7ed7095e5e87" />

<img width="1240" height="609" alt="Screenshot 2025-11-18 at 7 20 27 PM" src="https://github.com/user-attachments/assets/4f092f59-4d45-4248-a3cc-a0197dc57efe" />

<img width="484" height="398" alt="Screenshot 2025-11-18 at 7 20 49 PM" src="https://github.com/user-attachments/assets/9c2c89a2-7ded-403a-8b62-406e983730b8" />

AnthonyWeathers commented:
> Hi Anthony,
> 
> I have reviewed your PR locally. Although the whole functionality works as expected. I just have one concern, why do we have to click again on the Badges in the Tasks and Timelogs section for the notifications to disappear, why does the count not disappear when we click on the Badges Earned logo which then open the Badges in the Tasks and Timelogs section?


That is a fair question that I hadn't noticed occurred, I had assumed it would clear it, but I might not have checked that or left it like that at the time. I'll see if I can make it resolve for both of them regardless which method you use to clear the badge notification clears it for both.

one-community commented:
Thank you all, merging!
